executor/executor.go

```diff
@@ -188,9 +188,11 @@ func Next(ctx context.Context, e Executor, req *chunk.Chunk) error {
 		start := time.Now()
 		defer func() { base.runtimeStats.Record(time.Since(start), req.NumRows()) }()
 	}
-	sessVars := base.ctx.GetSessionVars()
-	if atomic.CompareAndSwapUint32(&sessVars.Killed, 1, 0) {
-		return ErrQueryInterrupted
+	if base.ctx != nil {
```

Comment:
When would `base.ctx` be `nil`?

Reply:
maybe some other `Executor`  forget to initial `baseExecutor` just like `ChangeExec `

Reply:
Shall we make it an error to leave `baseExecutor` out? It seems if it's `nil` then we are introducing something that cannot be interrupted by `KILL` here.

Reply:
removed, PTAL again

Reply:
How about adding a test for this case?